### DISCUSSION TITLE
Redundancy error when cloning the project and running the tests.

### DISCUSSION THREAD

**Discussion by Phssilva** [fatiando/community#144] [Q&A] [answered]
Hello, 
I need help.
I cloned the project and as usual, I always run the test before starting to edit the code. In this case, when the test is already failing due to redundancy, what do you recommend? I couldn't find any redundancy in the init.py file. Is there any related issue regarding this? Additionally, I noticed that in the project repository verde, in the same folder where _version.py exists, there is also another file named _version_generated, which I didn't notice in the ensaio project. Do you know anything about this? I had the same error in the boule project when I ran the tests.

This error only occurs in the projects 'boule' and 'ensaio'. I managed to clone the 'verde' project and submit a pull request with the tests working normally.

look the traceback:

'''
======================================================== test session starts ========================================================
platform linux -- Python 3.11.4, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/ph_silva/UFSC/OpenSource/boule
configfile: pyproject.toml
plugins: anyio-3.7.1
collected 0 items / 1 error                                                                                                         

============================================================== ERRORS ===============================================================
__________________________________________ ERROR collecting boule/tests/test_ellipsoid.py ___________________________________________
ImportError while importing test module '/home/ph_silva/UFSC/OpenSource/boule/boule/tests/test_ellipsoid.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../../miniconda3/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
../__init__.py:12: in <module>
    from ._version import __version__
../_version.py:13: in <module>
    from . import _version_generated
E   ImportError: cannot import name '_version_generated' from partially initialized module 'boule' (most likely due to a circular import) (/home/ph_silva/UFSC/OpenSource/boule/boule/__init__.py)
====================================================== short test summary info ======================================================
ERROR test_ellipsoid.py
'''

Sorry! It's my first open-source contribution, and I'm still learning.

**Comment by leouieda** [comment, marked as answer]
@Phssilva have you tried running `make install` before running the tests?

**Comment by Phssilva** [comment]
@leouieda I had downloaded the Verde project and managed to run the test successfully without running make install, so I assumed it was the same pattern for the others.

Thank you very much for the solution, I managed to run the tests for both the Ensaio and Boule projects, now I will continue with the tasks I set out to do.

**Reply by leouieda** [reply]
No problem! Glad it worked out.
